SCons/Action.py

```diff
@@ -924,7 +924,7 @@ def execute(self, target, source, env, executor=None):
 
         escape = env.get('ESCAPE', lambda x: x)
 
-        ENV = get_default_ENV(env)
+        ENV = env.get('SHELL_ENV_EXPANDER', get_default_ENV)(env, target, source)
 
         # Ensure that the ENV values are all strings:
         for key, value in ENV.items():
```

Comment:
I notice this does modify-in-place (not something introduced by this PR) - presumably that's okay? While `_subproc` creates a new dict for ENV.

Reply:
when you say in place, you mean its affecting the source Environment? thats not good, because in mongodb use case we need to have it expand out to different things based on target source. I might need to make sure my test covers that.

Reply:
My reading is, in the normal case `get_default_ENV` just returns the reference: `return env['ENV']`, so when you modify it, that's what is being modified.  May not be any issue, just wanted to ask.

Reply:
it may be of concern that get_default_ENV return the direct reference and so it would be modified after an action ran. Bit unwieldy behavior IMHO